doc/source/examining/loading_data.rst

```diff
@@ -1661,6 +1661,41 @@ The ``load_particles`` function also accepts the following keyword parameters:
 ``bbox``
        The bounding box for the particle positions.
 
+A novel use of the ``load_particles`` function is to facilitate SPH
+visualization of non-SPH particles. See the example below:
+
+.. code-block:: python
+
+    import yt
+
+    # Load dataset
+    ds = yt.load('FIRE_M12i_ref11/snapshot_600.hdf5')
+
+    # Reload star particles into a stream dataset
+    ad = ds.all_data()
+    pt = 'PartType4'
+    fields = ['particle_mass'] + [f'particle_position_{ax}' for ax in 'xyz']
+    data = {('io', field): ad[pt, field] for field in fields}
+    ds_star = yt.load_particles(data, data_source=ad)
```

Comment:
So what is exactly happening here? Do the "star" fields become "gas" fields in `ds_star`? If so, that seems a bit confusing to me.

Reply:
It looks like they're just `"io"`, I guess.

Reply:
Yes, the particle type becomes `'io'`, which is specified in the data dictionary. But this is just an arbitrary choice following the convention that `'io'` is the SPH particle type. Later in `ds_star.add_sph_fields()`, `sph_ptype='io'` is assumed as the default. But it could be set to other value, `ds_star.add_sph_fields(sph_ptype='star')` for example.

Reply:
Ok, this is fine then. Thanks for the clarification!

Reply:
So, I can't help wondering if there is a way to encapsulate these steps into a single function:

```python
data = {('io', field): ad[pt, field] for field in fields}
ds_star = yt.load_particles(data, data_source=ad)
```
Something where you'd pass in `pt` and `fields` and then this would happen under the hood.

Reply:
Thanks for bringing this up. I have thought about that in a [previous design](https://github.com/yt-project/yt/pull/2186/commits/d00a96efe022cd66e36345fc43ee4976b7c84635#diff-fc0105fc04a164b7cdc4e4fc59b91064L1336), but later turned away. The main reason is I haven't figured out how to do it without adding a new function. And I'm not convinced this use case is common enough to justify a special function.

There is probably an elegant way to do it. I'd like to have more discussions about this. @MatthewTurk @brittonsmith @ngoldbaum might also have some perspectives.

Reply:
The thing is that the `add_sph_fields` method is new, but I can only see it realistically being used for this use case (feel free to object, however, as I could be missing something). So I wonder if `add_sph_fields` should really be `_add_sph_fields` and there is a method that hangs off the original dataset which creates the new stream dataset and calls `_add_sph_fields` at the end.

Reply:
I think it's good that it's public. It should be possible to do this with an arbitrary N-body particle dataset, it doesn't necessarily need to be done via `load_particles` I think.

I also think making this project too magical might be brittle or hard to extend, it's good that it's sort of manual right now as it forces the user to think about what they're doing.

Reply:
That's fair.

Reply:
This functionality will be included in my current PR #2187 when this one gets merged.

Reply:
I think this is settled for now. I'll stay with the current design.